clang/test/ClangScanDeps/diagnostics.c

```diff
@@ -34,8 +34,8 @@ module mod { header "mod.h" }
 // CHECK:            ],
 // CHECK-NEXT:       "context-hash": "[[HASH_MOD:.*]]",
 // CHECK-NEXT:       "file-deps": [
-// CHECK-NEXT:         "[[PREFIX]]/mod.h"
-// CHECK-NEXT:         "[[PREFIX]]/module.modulemap"
+// CHECK-DAG:          "[[PREFIX]]/mod.h"
```

Comment:
We already have a test that diffs two scan results (`submodule-order.c`), so I believe that would catch non-deterministic ordering. (We might need to tweak the test so that one of the modules contains multiple files.)

Reply:
This test has very few files and isn't making any attempt to be exhaustive about which kinds of input files it has. I don't think it provides very good coverage for checking the order is deterministic.